src/contracts/GPv2Settlement.sol

```diff
@@ -290,6 +290,25 @@ contract GPv2Settlement {
         filledAmount[uid] = currentFilledAmount;
     }
 
+    /// @dev Allows settlment function to make arbitrary contract executions.
+    /// @param interaction contains address and calldata of the contract interaction.
+    function executeInteraction(GPv2Encoding.Interaction memory interaction)
+        internal
+        returns (bytes memory response)
+    {
+        // To prevent possible attack on user funds, we explicitly disable
+        // interactions with AllowanceManager contract.
+        require(
+            interaction.target != address(allowanceManager),
+            "GPv2: forbidden interaction"
+        );
+        bool success;
+        // solhint-disable-next-line avoid-low-level-calls
+        (success, response) = (interaction.target).delegatecall(
```

Comment:
Why are we using a delegate call here? Do we want the code of the interaction be executed in the settlement contract's state? I think this should rather be a normal call.

Reply:
This should be a normal `CALL`. For some reason I had remembered wrong and told @bh2smith it was a `DELEGATECALL` :see_no_evil: .

Reply:
```suggestion
        (success, response) = (interaction.target).call(
```